libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Button.kt

```diff
@@ -133,44 +133,59 @@ internal fun ButtonInternal(
         ButtonSize.Large -> 48.dp
     }
 
-    val contentPadding = when (size) {
-        ButtonSize.Medium -> {
-            when (style) {
-                ButtonStyle.Text -> PaddingValues(horizontal = 12.dp, vertical = 10.dp)
-                else -> PaddingValues(horizontal = 16.dp, vertical = 10.dp)
-            }
+    val hasNoStartDrawable = !showProgress && leadingIcon == null
+
+    val (paddingStart, paddingCenter, paddingEnd) = when (size) {
```

Comment:
`paddingCenter` is always 0 with no start drawable and 8 with one, couldn't we extract that from the `when` to simplify it? That way you could just return a `PaddingValues` with start and end values.

Reply:
If you look at them side by side it is easier to understand and follow the code, but to someone who just came to the code it'll probably look a lot more convoluted (at least that was my case). Also, if the structure of the designs change in the future (something added, something moved) we'd have to change the code again to match it.